pkg/templates/encryption.go

```diff
@@ -159,7 +159,11 @@ func pkcs7Unpad(paddedValue []byte) ([]byte, error) {
 // processEncryptedStrs replaces all encrypted strings with the decrypted values. Each decryption is handled
 // concurrently and the concurrency limit is controlled by t.config.DecryptionConcurrency. If a decryption fails,
 // the rest of the decryption is halted and an error is returned.
-func (t *TemplateResolver) processEncryptedStrs(options *ResolveOptions, templateStr string) (string, error) {
+func (t *TemplateResolver) processEncryptedStrs(
+	options *ResolveOptions,
+	templateResult *TemplateResult,
+	templateStr string,
+) (string, error) {
```

Comment:
Did you consider/try having this and `getOrList` return an additional value reporting whether there was sensitive data? My instinct is that passing in a pointer, and mutating that parameter, could lead to problems later.

But I think I see why not having to juggle additional returned values in `ResolveTemplate` is kind of nice...

Reply:
`getOrList` is called by calling a template function such as `fromSecret` by the `text/template` package, so I don't have a choice there to handle it in `ResolveTemplate`. `processEncryptedStrs` is called explicitly by `ResolveTemplate`, so for that one, I could handle it in `ResolveTemplate`.

Do you have a strong preference?

Reply:
If any of them need to do it this way, it doesn't make sense to mix it up. So this is fine.